src/System.Management.Automation/security/Authenticode.cs

```diff
@@ -289,147 +293,108 @@ internal static Signature GetSignature(string fileName, string fileContent)
-                                        Signature.CatalogApiAvailable = false;
-                                    }
-                                    else
-                                    {
-                                        // ProductFile has to be Catalog signed. Hence validating
-                                        // to see if the Catalog API is functional using the ProductFile.
-                                        Signature productFileSignature = GetSignatureFromCatalog(productFile);
-                                        Signature.CatalogApiAvailable = (productFileSignature != null && productFileSignature.Status == SignatureStatus.Valid);
-                                    }
-                                }
-                            }
-                        }
-                        else
-                        {
-                            // If calling NativeMethods.WTGetSignatureInfo failed (returned a non-zero value), we still want to set Signature.CatalogApiAvailable to false.
-                            Signature.CatalogApiAvailable = false;
-                        }
-                    }
-                    finally
-                    {
-                        if (phStateData != IntPtr.Zero)
-                        {
-                            NativeMethods.FreeWVTStateData(phStateData);
-                        }
+            if (fileSigInfo.SigningCertificate is null)
+            {
+                signature = new Signature(filename, error);
+            }
+            else
+            {
+                signature = fileSigInfo.TimestampCertificate is null ?
+                    new Signature(filename, error, fileSigInfo.SigningCertificate) :
+                    new Signature(filename, error, fileSigInfo.SigningCertificate, fileSigInfo.TimestampCertificate);
+            }
 
-                        if (ppCertContext != IntPtr.Zero)
-                        {
-                            NativeMethods.CertFreeCertificateContext(ppCertContext);
-                        }
-                    }
-                }
+            switch (fileSigInfo.Kind)
+            {
+                case SignatureKind.None:
+                    signature.SignatureType = SignatureType.None;
+                    break;
+
+                case SignatureKind.Embedded:
+                    signature.SignatureType = SignatureType.Authenticode;
+                    break;
+
+                case SignatureKind.Catalog:
+                    signature.SignatureType = SignatureType.Catalog;
+                    break;
+
+                default:
+                    System.Diagnostics.Debug.Fail("Signature type can only be None, Authenticode or Catalog.");
+                    break;
             }
-            catch (TypeLoadException)
+
+            signature.IsOSBinary = fileSigInfo.IsOSBinary;
+
+            if (signature.SignatureType == SignatureType.Catalog && !Signature.CatalogApiAvailable.HasValue)
```

Comment:
Would you want the CatalogApiAvailable.HasValue property to be gotten for the `Signature signature` instance created on L336, or on the type?
```suggestion
            if (signature.SignatureType == SignatureType.Catalog && !signature.CatalogApiAvailable.HasValue)
```